server/game/core/card/Card.ts

```diff
@@ -455,11 +477,79 @@ export class Card extends OngoingEffectSource {
         this.game.registerMovedCard(this);
     }
 
+    public initializeZone(zone: Zone) {
+        Contract.assertIsNullLike(this._zone, `Attempting to initialize zone for card ${this.internalName} to ${zone.name} but it is already set`);
+
+        this._zone = zone;
+
+        this.initializeForStartZone();
+        this.initializeForCurrentZone(null);
+    }
+
+    // eslint-disable-next-line @typescript-eslint/no-empty-function
+    protected initializeForStartZone(): void {}
+
+    private addSelfToZone(zoneName: MoveZoneName) {
+        switch (zoneName) {
+            case ZoneName.Base:
+                this._zone = this.owner.baseZone;
+                Contract.assertTrue(this.isLeader());
```

Comment:
Would the base itself get into this block?

Reply:
With the current architecture the base is never actually moved, it is initialized already inside the base zone